server/metric/mock_metric_test.go

```diff
@@ -253,7 +298,112 @@ func TestMockMetricServer_GetMonitoredResourceDescriptorError(t *testing.T) {
 	}
 }
 
-func TestMockMetricServer_DeleteMetricDescriptorError(t *testing.T) {
+func TestMockMetricServer_GetMetricDescriptor_NotFoundError(t *testing.T) {
+	setup()
+	defer tearDown()
+
+	in := &monitoring.GetMetricDescriptorRequest{
+		Name: "test",
+	}
+	want := validation.StatusMetricDescriptorNotFound
+	response, err := client.GetMetricDescriptor(ctx, in)
+	if err == nil {
+		t.Errorf("GetMetricDescriptor(%q) == %q, expected error %q", in, response, want.Message())
+	}
+
+	if s := st.Convert(err); s.Code() != want.Code() {
+		t.Errorf("GetMetricDescriptor(%q) returned error %q, expected error %q",
+			in, s.Message(), want.Message())
+	}
+}
+
+func TestMockMetricServer_DeleteMetricDescriptor_NotFoundError(t *testing.T) {
+	setup()
+	defer tearDown()
+
+	in := &monitoring.DeleteMetricDescriptorRequest{
+		Name: "test",
+	}
+	want := validation.StatusMetricDescriptorNotFound
+	response, err := client.DeleteMetricDescriptor(ctx, in)
+	if err == nil {
+		t.Errorf("DeleteMetricDescriptor(%q) == %q, expected error %q", in, response, want.Message())
+	}
+
+	if s := st.Convert(err); s.Code() != want.Code() {
+		t.Errorf("DeleteMetricDescriptor(%q) returned error %q, expected error %q",
+			in, s.Message(), want.Message())
+	}
+}
+
+func TestMockMetricServer_MetricDescriptor_DataRace(t *testing.T) {
+	setup()
+	defer tearDown()
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+		_, err := client.CreateMetricDescriptor(ctx, &monitoring.CreateMetricDescriptorRequest{
+			Name:             "test-create-metric-descriptor",
+			MetricDescriptor: &metric.MetricDescriptor{Name: "test-metric-descriptor-1"},
+		})
+		if err != nil {
+			t.Fatalf("failed to call CreateMetricDescriptor: %v", err)
+		}
+	}()
+
+	go func() {
```

Comment:
I think you can just do one `go func()` and run the second call in the original goroutine, since right now the original goroutine isn't really doing anything

Reply:
Yeah, I was mistaken in thinking the goroutines were separate threads when a goroutine in of itself is a lightweight thread.